### DISCUSSION TITLE
Trim in publishing

### DISCUSSION THREAD

**Discussion by taooceros** [Flow-Launcher/Flow.Launcher#518] [Ideas]
**Question**
Why don't we trim in publishing profile? In dotnet 5 environment, although it takes time to do the job, it can reduce the output size from 92MB to about 58MB

@JohnTheGr8 @jjw24

**Comment by jjw24** [comment]
That would be good to trim down the size, I just haven't had time to look into it

**Comment by JohnTheGr8** [comment]
Trimming is still experimental afaik.

Also, curiously, self-contained `net5.0-windows` output is actually larger than the `netcoreapp3.1` equivalent: 84mb vs 78mb for the executable...

**Comment by taooceros** [comment]
> Trimming is still experimental afaik.
> 
> Also, curiously, self-contained `net5.0-windows` output is actually larger than the `netcoreapp3.1` equivalent: 84mb vs 78mb for the executable...

Maybe that is because of the large WinRT runtime. Around 26mb.

**Comment by taooceros** [comment]
hmmm seems that trimming break Flow to work as intended. Need more testing.

**Comment by jjw24** [comment]
A couple more suggestions to try, leaving it here for future reference:
https://ianqvist.blogspot.com/2018/01/reducing-size-of-self-contained-net.html

https://medium.com/@singhsukhpinder/how-to-reduce-net-core-app-size-90045757cd99

https://stackoverflow.com/questions/48009976/optimizing-net-core-self-contained-deployment

**Comment by taooceros** [comment]
Why I cannot see your message even in here.....😶

**Reply by XavBryant** [reply]
Had some trouble.
A version that does exist does work sort of. Control panel plugin does not work, and portable mode is a ??

**Reply by taooceros** [reply]
> Had some trouble.
> A version that does exist does work sort of. Control panel plugin does not work, and portable mode is a ??

Maybe plugin shouldn't be trimmed. By the way, you may need the self-contained=true as well for testing this.
Try take a look on the post_build.ps1, and builds other projects with release, and publish only the Flow.Launcher project.

**Reply by XavBryant** [reply]
I dont know how to use that file.

**Reply by taooceros** [reply]
hmm just try only publish the Flow.Launcher project, and build other project before it. Don't publish everything as self-contained. Sorry, I got something to do now, so unable to help you until later.

**Reply by XavBryant** [reply]
Working publish build is at 80mb after stripping plugins. Half the size of the original.
Cant recreate however.

--update1. Exported all of my current settings over, and the application seems completely stable. Final size compared to my older version: 74.51mb [this includes deleting all plugins except plugin manager, shell, and calculator]

--update2: still trying to recreate this build. 
link source:https://github.com/dotnet/standard/blob/release/2.0.0/Microsoft.Packaging.Tools.Trimming/docs/trimming.md#roots
i forgot to include   <TrimUnusedDependencies>true</TrimUnusedDependencies> for both the build and publish.

**Comment by XavBryant** [comment]
### Resolved.
Install https://github.com/dotnet/standard/blob/release/2.0.0/Microsoft.Packaging.Tools.Trimming
add <TrimUnusedDependencies>true</TrimUnusedDependencies> onto 5.0SelfContained.pubxml

seems to work fine.
**from 293mb to 127mb, mileage may vary but no issues encountered so far.**

**Reply by taooceros** [reply]
Looks quite promising. You can open a pull request for that😲
